soh/CMakeLists.txt

```diff
@@ -0,0 +1,1765 @@
+			STORMLIB_NO_AUTO_LINK
+		)
+	endif()
+endif()
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+	target_compile_definitions(${PROJECT_NAME} PRIVATE
+		"$<$<CONFIG:Debug>:"
+				"_DEBUG"
+		">"
+		"$<$<CONFIG:Release>:"
+			"NDEBUG"
+		">"
+		"SPDLOG_ACTIVE_LEVEL=0;"
+		"_CONSOLE;"
+		"_CRT_SECURE_NO_WARNINGS;"
+		"ENABLE_OPENGL;"
+		"UNICODE;"
+		"_UNICODE"
+	)
+endif()
+################################################################################
+# Compile and link options
+################################################################################
+if(MSVC)
+    if("${CMAKE_VS_PLATFORM_NAME}" STREQUAL "x64")
+        target_compile_options(${PROJECT_NAME} PRIVATE
+            $<$<CONFIG:Debug>:
+                /sdl-;
+                /w
+            >
+            $<$<CONFIG:Release>:
+                /Oi;
+                /sdl;
+                /Gy;
+                /W3
+            >
+            /permissive-;
+            /MP;
+            ${DEFAULT_CXX_DEBUG_INFORMATION_FORMAT};
+            ${DEFAULT_CXX_EXCEPTION_HANDLING}
+        )
+    elseif("${CMAKE_VS_PLATFORM_NAME}" STREQUAL "Win32")
+        target_compile_options(${PROJECT_NAME} PRIVATE
+            $<$<CONFIG:Debug>:
+                /RTCs
+            >
+            $<$<CONFIG:Release>:
+                /O2;
+                /Oi;
+                /Gy
+            >
+            /permissive-;
+            /MP;
+            /sdl-;
+            /w;
+            ${DEFAULT_CXX_DEBUG_INFORMATION_FORMAT};
+            ${DEFAULT_CXX_EXCEPTION_HANDLING}
+        )
+    endif()
+    if("${CMAKE_VS_PLATFORM_NAME}" STREQUAL "x64")
+        target_link_options(${PROJECT_NAME} PRIVATE
+            $<$<CONFIG:Debug>:
+                /INCREMENTAL
+            >
+            $<$<CONFIG:Release>:
+                /OPT:REF;
+                /OPT:ICF;
+                /INCREMENTAL:NO
+            >
+            /DEBUG;
+            /SUBSYSTEM:CONSOLE;
+            /FORCE:MULTIPLE
+        )
+    elseif("${CMAKE_VS_PLATFORM_NAME}" STREQUAL "Win32")
+        target_link_options(${PROJECT_NAME} PRIVATE
+            $<$<CONFIG:Debug>:
+                /STACK:8777216
+            >
+            $<$<CONFIG:Release>:
+                /OPT:REF;
+                /OPT:ICF;
+                /INCREMENTAL:NO;
+                /FORCE:MULTIPLE
+            >
+            /DEBUG;
+            /SUBSYSTEM:CONSOLE
+        )
+    endif()
+endif()
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+	target_compile_options(${PROJECT_NAME} PRIVATE
+		-Wall -Wextra -Wno-error
+        	-Wno-unused-parameter
+        	-Wno-unused-function
+        	-Wno-unused-variable
+		-Wno-missing-field-initializers
+        	-Wno-parentheses
+        	-Wno-narrowing
+        	$<$<COMPILE_LANGUAGE:CXX>:-Wno-deprecated-enum-enum-conversion>
+                -pthread
+		-msse2 -mfpmath=sse
+	)
+
+       target_link_options(${PROJECT_NAME} PRIVATE
+                -pthread
+                -Wl,-export-dynamic
```

Comment:
`-export-dynamic` does not exists on Darwin -- in cmake how can we exclude things depending on `uname -m`? This should only be added for non darwin platform. if you're on SoH discord, you can reach out to me @dcvz